scripts/fbi/hate_crime/geo_id_resolver.py

```diff
@@ -0,0 +1,175 @@
+from FBI hate crime dataset
+
+"""
+
+import os
+import io
+import sys
+import csv
+
+from .manual_geocodes import MANUAL_COUNTY_MAP
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_SCRIPT_PATH,
+                             '../../../util'))  # For State and County map
+from alpha2_to_dcid import USSTATE_MAP
+from county_to_dcid import COUNTY_MAP
+
+city_geocodes_csv_path = os.path.join(_SCRIPT_PATH,
+                                      '../crime/city_geocodes.csv')
+
+_US_GEO_CODE_UPDATE_MAP = {
+    # Replacing/Updating State abbreviations given in the data. Required in case of wrong state abbr mentioned in the data.
+    'NB': 'NE',  # Nebraska State
+    'GM': 'GU'  # Guam State
+}
+
+_IGNORE_STATE_ABBR = ['FS']  # Federal mentioned as State in data
+
+_IGNORE_COUNTY_SUFFIX = [
+    'Unified Police Department', 'Police Department', 'Public Safety'
+]
+
+city_geo_codes = {}
+with open(os.path.join(city_geocodes_csv_path), encoding="utf8") as csvfile:
+    csv_reader = csv.reader(csvfile)
+    for a in csv_reader:
+        city_geo_codes[a[0]] = a[1]
+
+
+def update_state_abbr(state_abbr):
+    if state_abbr in _US_GEO_CODE_UPDATE_MAP:
+        state_abbr = _US_GEO_CODE_UPDATE_MAP[state_abbr]
+    return state_abbr
+
+
+def get_city_variants(city):
+    city = city.strip().lower()
+    city_variants = [city]
+
+    # for delim in [",", "-"]:
+    #     if delim in city:
+    #         city_variants.extend(city.split(delim))
+
+    city_variants.extend([
+        city + " city", city + " village",
+        city.removesuffix("city"),
+        city.removesuffix("village"),
+        city.removesuffix("town"),
+        city.removesuffix("township"),
+        city.removesuffix("borough")
+    ])
+
+    return city_variants
+
+
+def read_city_geocodes():
+    """ Read geo codes from city_geocodes """
+    city = {}
+    with open(os.path.join(city_geocodes_csv_path), encoding="utf8") as csvfile:
+        csv_reader = csv.reader(csvfile)
+        for a in csv_reader:
+            city[a[0]] = a[1]
+    return city
+
+
+def city_to_dcid(state_abbr, city):
+    city_valid_names = get_city_variants(city)
+    city_state_list = [
+        city.strip().lower() + " " + state_abbr.lower()
+        for city in city_valid_names
+    ]
+
+    for city_state in city_state_list:
+        if city_state in city_geo_codes:
+            return "geoId/" + city_geo_codes[city_state]
+    return ''
+
+
+def preprocess_county(county):
+    for suffix in _IGNORE_COUNTY_SUFFIX:
+        if suffix in county.strip():
+            return county.removesuffix(suffix).strip()
+    return county.strip()
+
+
+def get_county_variants(county):
+    county_variants = [
+        county,
+        county + " County",
+        county.removesuffix('County'),
+        county + " Parish",
+    ]
+    return county_variants
+
+
+def county_to_dcid(state_abbr, county):
+
```

Comment:
can these remove extra lines between single statements.

is 'pylint' or 'yapf  --style google' adding these?

Reply:
I will check again

Reply:
run_test.sh -f (fixing lint using yapf --style google) is adding these extra lines